File: src/features/typing/data/go/merge_intervals.go
package main

import "sort"

func merge(intervals [][]int) [][]int {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})
	result := [][]int{}
	for _, interval := range intervals {
		n := len(result)
		if n == 0 || result[n-1][1] < interval[0] {
			result = append(result, interval)
			continue
		}
		if interval[1] > result[n-1][1] {
			result[n-1][1] = interval[1]
		}
	}
	return result
}

File: src/features/typing/data/ts/merge_intervals.ts
function merge(intervals: number[][]): number[][] {
  intervals.sort((a, b) => a[0] - b[0]);
  const result: number[][] = [];
  for (const interval of intervals) {
    const last = result[result.length - 1];
    if (last === undefined || last[1] < interval[0]) {
      result.push(interval);
      continue;
    }
    last[1] = Math.max(last[1], interval[1]);
  }
  return result;
}
